docs/proposal-cylc-rose-installing-rose-configs.md

```diff
@@ -0,0 +1,291 @@
+# Cylc-Rose - Installation of Rose Configs
+
+## Rose2019 & Cylc7
+
+The `rose suite-run` command loads in the `rose-suite.conf` file and performs
+certain actions based on its configuration.
+
+There are various ways to meddle with the configuration as it is written. Here
+are the different methods for defining the Rose configuration in the order
+in which they are applied:
+
+1. The "base" `rose-suite.conf` configuration.
+2. Optional configurations defined by:
+   * The `rose-suite.conf|opts` configuration.
+   * The `ROSE_SUITE_OPT_CONF_KEYS` environment variable.
+   * The `-O, --opt-conf-key` CLI option.
+   Note these are all additive so can be used in combination
+3. CLI "defines"
+   * Suite template defines via the `-S, --define-suite` CLI option.
+   * Any Rose configuration via the `-D, --define` CLI option.
+
+> **Note:** The order in which opt conf keys are applied appears to be
+> undocumented but the answer lies in the code
+> [here](https://github.com/metomi/rose/blob/99ad9e766571212c6c3afdbe3915c75ba05e6c68/metomi/rose/run.py#L149-L155)
+> and
+> [here](https://github.com/metomi/rose/blob/99ad9e766571212c6c3afdbe3915c75ba05e6c68/metomi/rose/config.py#L1280-L1286).
+
+
+`rose suite-run` copies source files into the run directory *excluding* the
+`rose-suite.conf` file.
+
+The parsed Rose config is dumped in the run directory into
+`log/rose-suite.conf` for logging reasons, however, it is not read by Rose
+from this location for any further purpose.
+
+Any configuration set by either (2) or (3) above must be provided
+in any subsequent `rose suite-run` command or it will be lost. This includes
+`rose suite-run --reload` and `rose suite-run --restart` making these
+commands especially dangerous.
+
+## Rose2 & Cylc8
+
+The `rose suite-run` command is better thought of as `rose suite-install --run`.
+By default it runs the suite but can also `--reload` or `--restart`
+on request.
+
+In the new system `rose suite-run` has been replaced by `cylc install`
+which will call out to the new `cylc-rose` plugin for Rose specific
+functionality.
+
+This new command will be responsible for the installation of the Rose
+configuration into the run directory as well as supporting the command line
+options and environment variables that can change the Rose configuration
+(as outlined above).
+
+```
+usage: cylc install [opts]
+
+Plugins:
+  cylc-rose:
+    Opts:
+      -S, --define-suite, (--define-template-var)
+        Define a Cylc Jinja2/EmPy template variable.
+      -D, --define
+        Define a Rose configuration.
+      -O, --opt-conf-key
+        Define a Rose optional configuration key.
+    Env:
+      ROSE_SUITE_OPT_CONF_KEYS
+        Define a set of Rose optional configuration keys to append to the list.
+```
+
+## Proposed Changes
```

Comment:
The other key change is that the `suite.rc` / `cylc.flow` file will be installed unchanged, unlike at present.

Reply:
Not actually a change for Cylc8.

Reply:
Fair enough. I was just thinking that one of the reasons we have to install the `rose-suite.conf` file is that it will actually be read by `cylc play` whereas this isn't necessary at the moment because of the fact we modify `suite.rc`.